src/dodal/devices/motors.py

```diff
@@ -59,3 +61,24 @@ def __init__(
             self.phi = Motor(prefix + infix[4])
             self.omega = Motor(prefix + infix[5])
         super().__init__(name=name, prefix=prefix, infix=infix[0:3])
+
+
+class MotorGroup(StandardReadable):
+    """
+    Generic group of motors, dynamically created from a name -> PV dictionary
+    """
+
+    def __init__(self, motor_name_to_pv: dict[str, str], name=""):
+        with self.add_children_as_readables():
+            for raw_motor_name, pv in motor_name_to_pv.items():
+                motor_name = safe_identifier(raw_motor_name)
```

Comment:
You don't need to set the motor name- if a device's name is `""`, it will be named after it's parent-

```python
m = MotorGroup(name="foo")
m.x.name = "foo-x"
m.y.name = "foo-y"
```

Reply:
And devices that are created with `@device_factory()` will take the name of the function if their names are `""`